app/controllers/katello/api/v2/host_subscriptions_controller.rb

```diff
@@ -43,6 +44,40 @@ def destroy
       respond_for_destroy(:resource => @host)
     end
 
+    api :POST, "/hosts/subscriptions/", N_("Register a host with subscription and information.")
+    param :name, String, :desc => N_("Name of the host"), :required => true
```

Comment:
there's a current issue (http://projects.theforeman.org/issues/12383) open around the sub-man register call with --name as its currently ignored in favor of the hostname fact.  We'll probably have to wait for user feedback to know what the proper way to deal with this is.  I'll make a change to use the same logic here.